### PR TITLE
Gas price function override

### PR DESCRIPTION
**Pull request by epheph**
```
augur.setGasPriceFunction( (callback) => callback(123456789))
```

### CODE REVIEW

File: src/api/bind-contract-function.js [hunk 1]
```diff
@@ -19,7 +19,7 @@ function bindContractFunction(functionAbi) {
         }
         if (!isFunction(params[params.length - 1])) return console.error("Callback required");
         var callback = params.pop();
-        return ethrpc.callContractFunction(payload, function (err, response) {
+        return self.rpc.callContractFunction(payload, function (err, response) {
```
Comment (bthaile) [inline src/api/bind-contract-function.js]: flash scripts fail at this line. ` return self.callContractFunction(payload, function (err, response) {...`

TypeError: self.callContractFunction is not a function

**Comment by bthaile** [comment]
On a different note, I think this code is fine for the impl. I was able to wire everything up in the ui.

```
  if (newGasPriceFunction === undefined) {
    this.getGasPrice = defaultGetGasPrice;
  } else {
    this.getGasPrice = newGasPriceFunction;
  }
```

**Comment by bthaile** [comment]
now getting this error using flash scripts to push-timestamp
```
/Users/bthaile/gitrepos/augur.js/src/api/bind-contract-function.js:41
    var transact = function () { self.rpc.transact(payload, signer, accountType, onSent, onSuccess, onFailed); };
                                          ^

TypeError: Cannot read property 'transact' of undefined
    at transact (/Users/bthaile/gitrepos/augur.js/src/api/bind-contract-function.js:41:43)
    at Object.setTimestamp (/Users/bthaile/gitrepos/augur.js/src/api/bind-contract-function.js:49:5)
...
```

**Comment by epheph** [comment]
@bthaile you shouldn't need to do the undefined check now. Also, the transact should work.

**Comment by bthaile** [comment]
`flash push-timestamp -c 1 -w` gives same error:

```
/Users/bthaile/gitrepos/augur.js/src/api/bind-contract-function.js:41
    var transact = function () { self.rpc.transact(payload, signer, accountType, onSent, onSuccess, onFailed); };
                                          ^

TypeError: Cannot read property 'transact' of undefined
    at transact (/Users/bthaile/gitrepos/augur.js/src/api/bind-contract-function.js:41:43)
    at Object.setTimestamp (/Users/bthaile/gitrepos/augur.js/src/api/bind-contract-function.js:49:5)
    at /Users/bthaile/gitrepos/augur.js/scripts/flash/set-timestamp.js:17:30
    at /Users/bthaile/gitrepos/augur.js/scripts/flash/get-timestamp.js:20:7
    at /Users/bthaile/gitrepos/augur.js/src/api/bind-contract-function.js:26:11
    at /Users/bthaile/gitrepos/augur.js/node_modules/ethrpc/src/transact/call-contract-function.js:23:7
    at parseEthereumResponse (/Users/bthaile/gitrepos/augur.js/node_modules/ethrpc/src/decode-response/parse-ethereum-response.js:9:3)
    at responseHandler (/Users/bthaile/gitrepos/augur.js/node_modules/ethrpc/src/rpc/blockchain-message-handler.js:46:7)
    at /Users/bthaile/gitrepos/augur.js/node_modules/ethrpc/src/rpc/blockchain-message-handler.js:53:7
    at store.dispatch (/Users/bthaile/gitrepos/augur.js/node_modules/redux-thunk-subscribe/src/index.js:10:18)
```

**Pull request merged**